### ISSUE TITLE
Video starts playing right away, but subtitles take a few seconds to load

### ISSUE THREAD

**Issue by oddstr13** [jellyfin/jellyfin-kodi#35]
Seeking back to the start of the file once subtitles does load gives subtitles for the initial section also.

This may be an issue with how Jellyfin feeds us the data.

**Comment by mothe** [comment]
I have this same issue, and it only happens when using Kodi + Jellyfin as a client. Web and Android app work fine. Video starts playing immediately but subs take ~5-10 seconds to load on a direct stream. If I pause the video and wait the 10 seconds, they will appear immediately after resuming.

Jellyfin 10.3.7 (Official Docker Container)
Kodi 18.3 and 18.4
Jellyfin Kodi Addon 0.2.1
Tested on: ASS and PGS subtitle filestypes in MKV containers. Both experience the delay.

Not sure what else I can share to help,

**Comment by JustAMan** [comment]
I don't think this is a bug, more like how we expected it to behave... at least in web ui.
So that playback starts as fast as it can, and subtitles show when they're ready after extraction.

**Comment by oddstr13** [comment]
That can't be expected behaviour. That's like audio starting to play a few seconds before the video is ready.

I would rather wait an extra couple of seconds for subtitles to load properly, or at least have a config switch for it, if this is indeed a intentional choice

**Comment by leftl** [comment]
I am also having this issue, but only when using Kodi as a client. Web player and Android app work fine, as does direct playing the mkv container in Kodi via SMB. The loading delay only seems to happen with Jellyfin + Jellyfin-Kodi.
This would have to be an issue with the Jellyfin server though, right? Isn't this addon just building a local database of media served by Jellyfin, and Kodi is handling all of the streaming/decoding?

**Comment by barolo** [comment]
Also having such issue, with latest version of everything. Furthermore unburned, srt subtitles will desync as tome point when transcoding [ unsure if it's related to hw accelerated transcoding, as I'm not using unaccelerated one at all ]

**Comment by flashkiller999** [comment]
Can confirm the problem still exists.
Subtitle is delayed for about 5 seconds after video and audio already started playing.

Tested using a 1080p mkv with H264 (Bitrate: 2700 kb/s) and AAC, Subtitle format: SRT
Also tested using ASS/SSA: same behavior as with SRT.
Mode used: Direct play
Connection is a dedicated LAN connection (1 Gbits).

Problem does not occur when playing the file directly from a Samba share.
Problem does not occur on native Jellyfin-Android app and Webplayer. (Mode: Direct play)

OS: LibreELEC 9.2.5 (Kodi 18.8)
HW: Rasberry Pi 4 (2GB)
Jellyfin Addon version: 0.5.9+py2
Jellyfin Server: 10.6.4

**Comment by snortwolf** [comment]
Think I've tracked this delay down to `set_audio_subs()` in `player.py`.  Looks like subtitles actually start on video playback as expected, then effectively get reloaded, thus the visible delay.

All my subtitles are embedded, so I don't know how external subs work here, but commenting out the call to `set_audio_subs()` seems to fix this for me.

**Comment by mcarlton00** [comment]
I still haven't experience this personally, but from what you've said it seems as though it's (yet another) race condition.  I wonder if there would be a way to sync the external subs locally instead of needing to request them from the server.

**Comment by oddstr13** [comment]
So, the issue here seems to be a combination of a few things;
 - Kodi is not skipping back in a mkv to find the relevant chunk of subtitle data, nor keeping all subtitles in memory.
 - Neither we nor Kodi are checking if the requested track is the one already selected, thus discarding the subtitle data already loaded in memory. (I'm unsure if we have access to the information required to check this on our end)
 - The subtitle track is changed after the video has started playing. (even if it is to the one we where already playing)

Subtitle data is tiny compared to audio and video data.
When you change the audio track, the video will skip forward a little bit (a few frames it looks like), until Kodi finds another block of audio from the wanted track.
This is not done for subtitles, and it isn't desirable to do so either, as subtitles aren't there all the time.

Depending on whether the Jellyfin server actually provides any useful information on which track to use or not (it's possible it could potentially be storing the selected track between plays, or between episodes), the solution could indeed be to just remove that call to set_audio_subs().

**Comment by jasonmbrown** [comment]
Just wanted to mention that Im also experiencing this issue. Its most noticeable in anime, (Jap Audio, Eng Subtitles) For about 10-20 seconds after a video loads up subtitles are not visible If I pause and wait till they show up it seems to be fine. I haven't noticed any other issues but it would be nice to get this fixed.

**Comment by Weevild** [comment]
Still experiencing this as well, in the exact same way as previously described.

**Comment by NCLI** [comment]
Really annoying, and definitely still happening.

**Comment by Jucgshu** [comment]
Just dropping another "me too" comment, as it's been almost 2 years without any development, and the problem still occurs every single time I load a media file with subtitles.

Would greatly appreciate this issue gets more attraction.

**Comment by oddstr13** [comment]
Subtitles do instantly load, but seem to be replaced after a few seconds, possibly getting fetched as a separate track from the server, and triggering server side subtitle extraction before being handed back to Kodi.

I don't know for sure, as I haven't delved deeper into the issue.
Minor inconvenience/annoyance for me, so I have not yet prioritized dedicating time to it, but I *do* think about it from time to time.

**Comment by Jucgshu** [comment]
Thanks for your answer. I do no try to put pressure of any sort. Just bringing back a little light on this issue. Sure it's not a major thing, but from a user perspective, we do get video and audio instantly... and not subtitles. Which is really weird when it comes to a project as big as Jellyfin 😉

**Comment by rastarr** [comment]
OK, good to see this is not just me. Since I enable Thai subtitles for just about all media (Thai girlfriend), this tends to drive me crazy. I resort to having to reboot Kodi, once and maybe twice. I've yet to find any solution that works and it's hit and miss until the subtitles start to work.
I have a friend who mentioned the same problem with his installation with .srt files.

**Comment by fl0wfr** [comment]
By the way, when using path replacement, Jellyfin client seems to continue to load subtitles from Jellyfin, not from the NFS/SMB share it uses for loading the media, causing the display delay. It should use the embedded subtitles, or if not embedded subtitles, load the subtitles stored in the video folder.

**Comment by adolfotregosa** [comment]
Did anyone find a solution ? It was quickly noticeable in animes.

Edit: having configured native mode from the addon using path replacements this is no longer a problem

**Comment by rastarr** [comment]
> Did anyone find a solution ? It was quickly noticeable in animes.
> 
> Edit: having configured native mode from the addon using path replacements this is no longer a problem

well actually, and at least for me, I had JF configured to automatically d/l thai subtitles.
now since I do that myself, I discovered that disabling this feature fixed the probably and my subtitles play as expected

**Comment by jasonmbrown** [comment]
So the solution sadly is to pre-extract all your subtitle files because it
turns out jelly fin has to decode the entire file before it can pull the
subtitles out of the media (at least I had read this somewhere) but when
using separate subtitle files it can just send them immediately.

**Comment by adolfotregosa** [comment]
> So the solution sadly is to pre-extract all your subtitle files because it
> turns out jelly fin has to decode the entire file before it can pull the
> subtitles out of the media (at least I had read this somewhere) but when
> using separate subtitle files it can just send them immediately.
> […](#)

Native mode is working fine, no more delays showing the subtitles.

**Comment by tamodolo** [comment]
Well this is still an issue in 2025. 6 years on the go. I find incredible this pain isn't a thing for devs.

My case is a bit more complex. I use a combo of softwares to serve animes: First I create a metadata source with Shoko and read it with Jellyfin. Both runs direct on a TrueNAS server. The shokofin plugin builds a virtual file system to provide feature complete state. So native mode is painfully hard to make it work as I really don't know where I should point path replacements. Nor I have an example on how to do that. The only thing left for me is to use stream mode.

I'll be very gratefull for any help one can provide me! Really!

**Comment by bossanova808** [comment]
There is, at long long last, some hope for a proper fix for this issue.  This commit just made to Kodi 22: https://github.com/xbmc/xbmc/pull/27770 ..._finally_ seems to solve this issue.

(It's currently available in 'T4' test builds of CoreELEC from panni - and with those, subtitles always show up straight away...huzzah!).
